content/blog/deploying-the-infrastructure-of-oauth-server-for-cms-app/index.md

```diff
@@ -0,0 +1,371 @@
+
+We can extract what we need just for creating the certificate and delete irrelevant parts including uploading content to the S3 bucket and configure CloudFront. Then the rest only involves creating an east region provider, a certificate, and certificate validation.
+
+When we create the Alias Record, we configure it with application load balancer settings.
+
+```typescript
+// Creates a new Route53 DNS record pointing the domain to the CloudFront distribution.
+function createAliasRecord(
+    targetDomain: string, lb: awsx.elasticloadbalancingv2.ApplicationLoadBalancer): aws.route53.Record {
+    const domainParts = getDomainAndSubdomain(targetDomain);
+    const hostedZoneId = aws.route53.getZone({ name: domainParts.parentDomain }, { async: true }).then(zone => zone.zoneId);
+    return new aws.route53.Record(
+        targetDomain,
+        {
+            name: domainParts.subdomain,
+            zoneId: hostedZoneId,
+            type: "A",
+            aliases: [
+                {
+                    name: lb.loadBalancer.dnsName,
+                    zoneId: lb.loadBalancer.zoneId,
+                    evaluateTargetHealth: true,
+                },
+            ],
+        });
+}
+// Create the aliasRecord with targetdomain and application load balancer
+const aRecord = createAliasRecord(cmsStackConfig.targetDomain, alb);
+```
+
+We run `pulumi up` to deploy.
+
+### Github Workflow (Optional)
+
+We can automate our build with the Github workflow in the folder `cms-oauth/.github/workflows/build-and-deploy.yml`. The workflow uses the repository secret we set in Github.
+
+![Github Secret](./github_secret.jpg)
+
+## Last Step
+
+Finally, we need to update the CMS configuration file.
+
+To combine our deployment of CMS and CMS OAuth Server, we specify the site_domain and URL in `./public/config.yml` of the CMS folder.
+
+```yaml
+backend:
+  name: github
+  # Replace this with the Github repo you want to make change
+  repo: github-username/target-github-repo
+  # This site_domain and base_url are sudo domains
+  # Replace site_domain with targetDomain pulumi stack configuration inside cms/infrastructure folder
+  # Replace base_url with targetDomain pulumi stack configuration inside cms-oauth/infrastructure folder
+  site_domain: https://some-cms-domain.pulumi-demos.com
+  base_url: https://some-oauth-domain.pulumi-demos.com
+```
+
+The site_domain is the URL of the CMS and the base_url is the URL of the OAuth Server . They should be the same as the `targetDomain` variable we set in the stack configuration of both CMS and OAuth Server.
+
+## In the End
+
+Congratulations on building both CMS web application, the OAuth client-server, and deploy the infrastructure! Now if everything works perfectly, you would see the "Login with Github" button shown and click it would redirect to the Github login pages.
```

Comment:
```suggestion
Congratulations on building both CMS web application, the OAuth client-server, and also deploying the infrastructure! Now if everything works perfectly, you should see the **Login with Github** button and clicking it would redirect to the GitHub login page.
```